front/pages/api/w/[wId]/data_sources/tags.ts

```diff
@@ -0,0 +1,97 @@
+import { CoreAPI } from "@dust-tt/types";
+import { isLeft } from "fp-ts/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/auth_wrappers";
+import apiConfig from "@app/lib/api/config";
+import type { Authenticator } from "@app/lib/auth";
+import { getFeatureFlags } from "@app/lib/auth";
+import logger from "@app/logger/logger";
+import { apiError } from "@app/logger/withlogging";
+
+export const PostTagSearchBodySchema = t.type({
+  query: t.string,
+  queryType: t.string,
+  dataSources: t.array(t.string),
```

Comment:
Ideally here you should take an array of dataSourceViews instead (but can be fixed once core api is available)

Reply:
Yes I think I'll update when the endpoint is live as I'm likely gonna need to tweak things anyway!